### PR TITLE
FMC250M_4CH support

### PR DESCRIPTION
**Pull request by ericonr**
- [X] SI571
- [X] FMC Active Clock
- [x] FMC250M 4CH
- [x] FMC Common ADC
- [x] SPI
- [x] AD9510 (SPI)
- [X] ISLA (SPI) **this has been partially implemented. Only the basic register controls are available so far**

### CODE REVIEW

File: modules/si57x_ctrl.cc [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        .device_id = SI57X_CTRL_DEVID,
+        .abi_ver_major = 1
+    };
+}
+
+Core::Core(struct pcie_bars &bars):
+    RegisterDecoder(bars, ref_devinfo, {
+        PRINTER("STRP_COMPLETE", "Startup registers status", PrinterType::boolean, "valid", "invalid"),
+        PRINTER("CFG_IN_SYNC", "Registers synchronization status", PrinterType::boolean, "values in sync", "values NOT in sync"),
+        PRINTER("I2C_ERR", "I2C error status", PrinterType::boolean, "No errors", "I2C error ocurred"),
```
Comment (ericonr) [inline modules/si57x_ctrl.cc:23]: This seems to be flipped.

File: include/modules/si57x_ctrl.h [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+class Controller: public RegisterDecoderController {
+    std::unique_ptr<struct wb_si57x_ctrl_regs> regs_storage;
+    struct wb_si57x_ctrl_regs &regs;
+
+    Core dec;
+
+    /** The device's startup frequency. */
+    double fstartup;
+    /** The device's internal crystal frequency. */
+    double fxtal;
```
[the remark this answers is not in the document]
Comment (ericonr) [reply include/modules/si57x_ctrl.h:33]: It is, the constructor sets it.

File: modules/si57x_ctrl.cc (hunk 1 above)
Comment (ericonr) [reply modules/si57x_ctrl.cc:23]: It was indeed flipped.

File: modules/spi.cc [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    /* TODO: deal with commands spanning more than one register */
+    regs.x0 = 0;
+    memcpy(&regs.x0, wdata, wsize);
+    /* the bytes were in network order (MSB first), and were copied into x0, so
+     * we need to revert that */
+    regs.x0 = ntohl(regs.x0);
+    /* now the bytes are at the register's MSB, so we need to shift them to LSB */
+    regs.x0 >>= (4 - wsize) * 8;
+    /* finally, we need to make room for the response bytes */
+    regs.x0 <<= rsize * 8;
```
Comment (ericonr) [inline modules/spi.cc:144]: @henriquesimoes @gustavosr8 could you please take a look at this function to see if it's clear how the data is passed to the SPI core?

Comment (henriquesimoes) [reply modules/spi.cc:144]: It is clear enough, IMO.

Comment (gustavosr8) [reply modules/spi.cc:144]: Seems good to me!

**Pull request merged**